### DISCUSSION TITLE
docs: incorrect example code, bug: entity discovery

### DISCUSSION THREAD

**Discussion by lysol33** [mikro-orm/mikro-orm#5636] [General]
There is incorrect code example which is mixing CommonJS and ES module syntax: https://mikro-orm.io/docs/usage-with-js
More precisely "export const schema" part of the code

Also following this example I cannot achieve entity discovery in plain javascript. I'm not sure if I'm doing something wrong, but I cannot find help in the documentation.

 **./index.js:**
```
const { MikroORM } = require("@mikro-orm/postgresql");
const Author = require("./entities/Author.js");

async function init() {
  const orm = await MikroORM.init({
    dbName: "postgres",
    allowGlobalContext: true,
    entities: [Author],
  });
  const book = orm.em.create(Author, { title: "Book title" });
  orm.em.persistAndFlush(book);
}
init();
```

**./entities/Author.js**
```
const { Collection, EntitySchema } = require("@mikro-orm/core");
class Author {
  name;
  email;
  constructor(name, email) {
    this.name = name;
    this.email = email;
  }
}

const schema = new EntitySchema({
  class: Author,
  name: "Author",
  properties: {
    id: { primary: true, type: "number" },
    name: { type: "string" },
    email: { type: "string", unique: true },
    age: { type: "number", nullable: true },
  },
});

module.exports.Author = Author;
module.exports.entity = Author;
module.exports.schema = schema;
```

Error I get:
```
TypeError: Cannot read properties of undefined (reading 'name')
    at Utils.className 
```

**Comment by B4nan** [comment]
Hmm right, maybe its time to convert the JS examples to ESM completely.

Here is a working example app using JS entities with ESM:

https://github.com/mikro-orm/express-js-example-app

**Comment by B4nan** [comment]
> Also following this example I cannot achieve entity discovery in plain javascript. I'm not sure if I'm doing something wrong, but I cannot find help in the documentation.

If your code looks exactly like what you provided here, its because you export the entity as named export, but import it as if it would be a default export:

```diff
-const Author = require("./entities/Author.js");
+const { Author } = require("./entities/Author.js");
```

Also you should not specify the `name` option if you provide `class` option, but that's probably fine, the `class` should take precedence.

edit: this is btw one of the reasons to adopt TS, you made a mistake in your code that would be caught easily by it... :]
